### DISCUSSION TITLE
Chiral tags from SMILES and mol are different even after canonicalization

### DISCUSSION THREAD

**Discussion by iwyoo** [rdkit/rdkit#7483] [Q&A] [answered]
It is related with https://github.com/rdkit/rdkit/issues/2879.
I guess that there is a logic to fix chirality in `MolFromSmiles` method.
However, even if I get canonicalized SMILES representation, a mol object from it has different chiral tag with the original SMILES.
Is this intended feature? (or am I missing something?)

```python
import re

from rdkit import Chem

# SMILES with chiral notation
smiles = "[CH3:1][C@@H:2]1[C@H:3]([CH2:4][C@@H:5]2[CH2:6][C@@H:7]1[C:8]2([CH3:9])[CH3:10])[B:11]([Cl:12])[C@H:13]1[CH2:14][C@@H:15]2[CH2:16][C@H:17]([C@@H:18]1[CH3:19])[C:20]2([CH3:21])[CH3:22]"
mol = Chem.MolFromSmiles(smiles)

# After canonicalization, chiral notation in SMILES is stable and not changed anymore
smiles1 = Chem.MolToSmiles(mol, canonical=True)
smiles2 = Chem.MolToSmiles(Chem.MolFromSmiles(smiles1), canonical=True)
assert (smiles1 == smiles2)

# Get atoms with chiral tags from SMILES with regex [...@...:...]
chiral_atoms_from_smiles = re.findall(r"\[[^\]]*?@[^\]]*?:\d+\]", smiles1)

# Get chial atoms from mol by GetSmarts
chiral_atoms_from_mols = []
mol1 = Chem.MolFromSmiles(smiles1)
for atom in mol1.GetAtoms():
    if atom.GetChiralTag():
        # `GetSmarts` generates correct chiral tag.
        # print(atom.GetSmarts(), atom.GetChiralTag())
        chiral_atoms_from_mols.append(atom.GetSmarts())

# They are different!
print(chiral_atoms_from_smiles)
print(chiral_atoms_from_mols)
```

Following is the printed result.
```
['[C@@H:2]', '[C@@H:3]', '[C@H:13]', '[C@@H:15]', '[C@H:17]', '[C@@H:18]', '[C@@H:5]', '[C@@H:7]']
['[C@H:2]', '[C@@H:3]', '[C@@H:13]', '[C@H:15]', '[C@H:17]', '[C@H:18]', '[C@H:5]', '[C@H:7]']
```

**Comment by thegodone** [comment]
Hi,

Smiles chirality (@) represents a local chirality (ie tetrahedral rotation clockwise/anticlockwise) this is not a global definition for the molecule. Look at my article https://arxiv.org/pdf/2010.01027. So @@ is not
equal to R or S it depends of the rest of the molecule (same for @).

![image](https://github.com/rdkit/rdkit/assets/1186658/1809e010-bebf-44cf-acf6-1375eb8b4b6f)


Best

Guillaume

**Reply by iwyoo** [reply]
Thanks for your kind reply! That means `@@` and `@` in SMILES are not same as chiral tags in mol objects. I have one more question. In that context, `atom.GetSmarts` should properly reflect local tetrahedral rotation (not chiral tag), and can we consider the current behaviour of `GetSmarts` as a bug?

**Reply by thegodone** [reply, marked as answer]
There are 2 ideas:
- The fact that Smiles was used as a representation of a molecule was for database purpose only and Smarts is only a subset logic of Smiles.
- If you look at your hands image representation in a mirror are the same, even if your two hands are not the same. Both Smiles/Smarts were never created to be identical to a mirror projection. 
 
 
So you need to define a better representation that takes into account "R/S" tags instead of "@/@@", but you will see rapidly that it is not simple task to go back from R/S to @/@@ so SmilesToChiralSmiles is easy not the inverse for multiple chiral centers. 
So for me this is not a bug of Smart / Smiles this was never clearly identify as a weakness before my paper.
